parser_test.go

```diff
@@ -2426,6 +2426,15 @@ func (s *testParserSuite) TestDDL(c *C) {
 		{"create table nchar (a int, b nchar);", true, "CREATE TABLE `nchar` (`a` INT,`b` CHAR)"},
 		{"create table nchar (a int, b nchar(50));", true, "CREATE TABLE `nchar` (`a` INT,`b` CHAR(50))"},
 
+		// Test keyword `SERIAL`
+		{"create table t (a serial);", true, "CREATE TABLE `t` (`a` BIGINT UNSIGNED NOT NULL AUTO_INCREMENT UNIQUE KEY)"},
+		{"create table t (a serial null);", true, "CREATE TABLE `t` (`a` BIGINT UNSIGNED NOT NULL AUTO_INCREMENT UNIQUE KEY NULL)"},
```

Comment:
```suggestion
		{"create table t (a serial null);", true, "CREATE TABLE `t` (`a` BIGINT UNSIGNED AUTO_INCREMENT UNIQUE KEY)"},
```
this should be right

Reply:
The `Restore` method only require the restored sql is semantic equivalence to the origin, so  
I just make sure the semantic equivalence here,  leave the duplicate removal job behind.

The current codebase fail on this kind of sql as well (`create table t (a serial not null null not null null not null);` for example), Shall I fix this in this pr?

Reply:
I think it's TiDB's responsibility to remove duplication. @leoppro 
If we do this completely in the parser, tons of code in TiDB needs to be move.

Reply:
@kennytm what do you think about it

Reply:
Let's leave it as-is.